backend/app/services/metrics_collector.py

```diff
@@ -0,0 +1,95 @@
+# Copyright (C) 2025 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+import logging
+import time
+from datetime import UTC, datetime
+from multiprocessing import Lock, shared_memory
+from typing import NamedTuple
+from uuid import UUID
+
+import numpy as np
+
+from app.utils import Singleton
+
+logger = logging.getLogger(__name__)
+
+
+class LatencyMeasurement(NamedTuple):
+    """Individual latency measurement"""
+
+    model_id: str  # UUID as 36 character string "00000000-0000-0000-0000-000000000000"
+    latency_ms: float
+    timestamp: float
+
+
+class MetricsCollector(metaclass=Singleton):
+    """Process-safe metrics collector using shared memory for model latency data"""
+
+    SHM_NAME = "latency_metrics_shm"
+    MAX_MEASUREMENTS = 1024  # max number of measurements to keep
+    DTYPE = np.dtype(
+        [
+            ("model_id", "U36"),  # 36 * 4 = 144 bytes for UUID string
+            ("latency_ms", np.dtype(float)),  # 8 bytes for latency in milliseconds
+            ("timestamp", np.dtype(float)),  # 8 bytes for timestamp (epoch time in seconds)
+        ]
+    )  # 144 + 8 + 8 = 160 bytes per latency measurement
+    SIZE = DTYPE.itemsize * MAX_MEASUREMENTS  # 160 * 1024 = 163840 bytes (160KB) allocated
+
+    def __init__(self, max_age_seconds: int = 60):
+        self._max_age_seconds = max_age_seconds
+        self._lock = Lock()
+        try:
+            self._shm = shared_memory.SharedMemory(name=self.SHM_NAME)
+            self._array: np.ndarray = np.ndarray((self.MAX_MEASUREMENTS,), dtype=self.DTYPE, buffer=self._shm.buf)
+        except FileNotFoundError:
+            self._shm = shared_memory.SharedMemory(name=self.SHM_NAME, create=True, size=self.SIZE)
```

Comment:
Shared memory should be created centrally in the scheduler and passed to the process via arguments, similar to other shared resources. This way, we can cleanup it properly during application shutdown. Currently, shared memory is removed in a singleton that will be garbage collected twice, causing exception on the second attempt.